augur/filter.py

```diff
@@ -788,10 +788,18 @@ def apply_filters(metadata, exclude_by, include_by):
     for filter_function, filter_kwargs in exclude_by:
         # Apply the current function with its given arguments. Each function
         # returns a set of strains that passed the corresponding filter.
-        passed = metadata.pipe(
-            filter_function,
-            **filter_kwargs,
-        )
+        try:
+            passed = metadata.pipe(
+                filter_function,
+                **filter_kwargs,
+            )
+        except Exception as e:
+            if filter_function.__name__ == 'filter_by_query':
+                if type(e) == pd.core.computation.ops.UndefinedVariableError:
+                    raise AugurError(f"Query contains a column that does not exist in metadata.")
+                raise AugurError(f"Error when applying query. Ensure the syntax is valid per https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#indexing-query.")
+            else:
+                raise e
```

Comment:
Omitting the current exception when re-raising it:

```suggestion
                raise
```

…avoids muddying the stack trace with an extra line in the `except` block. Compare:

```
>>> try:
...     1/0
... except Exception as e:
...     raise
... 
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
ZeroDivisionError: division by zero
>>> try:
...     1/0
... except Exception as e:
...     raise e
... 
Traceback (most recent call last):
  File "<stdin>", line 4, in <module>
  File "<stdin>", line 2, in <module>
ZeroDivisionError: division by zero
```

This seems appropriate since the exception handler here is intended only to transform the error class/message of a specific kind and pass thru everything else unmodified.